src/operations/map/MapOperations.ts

```diff
@@ -115,47 +115,50 @@ export class MapOperations {
     const popup = new mapboxgl.Popup({ closeButton: false, closeOnClick: false });
 
     baseMap.on('mousemove', 'country-fills', (e: any) => {
-      const feature = e.features && (e.features[0] as CountryMapData);
+      const countryData = e.features && (e.features[0] as CountryMapData);
 
-      if (hoveredPolygonId && (!feature || hoveredPolygonId !== feature.id)) {
+      if (hoveredPolygonId && (!countryData || hoveredPolygonId !== countryData.id)) {
         baseMap.setFeatureState({ source: 'countries', id: hoveredPolygonId }, { hover: false });
         hoveredPolygonId = undefined;
         baseMap.getCanvas().style.cursor = '';
       }
 
-      if (!feature) {
+      if (!countryData) {
         popup.remove();
         return;
       }
 
       baseMap.getCanvas().style.cursor = 'pointer';
-      hoveredPolygonId = feature.id;
+      hoveredPolygonId = countryData.id;
       if (hoveredPolygonId) {
         baseMap.setFeatureState({ source: 'countries', id: hoveredPolygonId }, { hover: true });
       }
 
       switch (selectedMapType) {
         case 'ipc': {
-          const countryName = feature.properties.adm0_name;
-          const ipcColorMap = IPCMapOperations.generateColorMap(ipcData!, countries!);
+          const countryName = countryData.properties.adm0_name;
+          const ipcColorMap = IPCMapOperations.generateColorMap(ipcData, countries);
 
           if (!ipcColorMap[countryName]) {
             popup.remove();
             return;
           }
 
-          const { date_of_analysis, analysis_period, ipc_percent, ipc_popnmbr } = IPCMapOperations.findIpcData(
-            countryName,
-            ipcData!
-          );
+          const selectedCountryData = IPCMapOperations.findIpcData(countryName, ipcData);
+          const date_of_analysis = selectedCountryData?.date_of_analysis || 'N/A';
+          const analysis_period = selectedCountryData?.analysis_period || 'N/A';
+          const ipc_percent = selectedCountryData?.ipc_percent ?? 0;
+          const ipc_popnmbr = selectedCountryData?.ipc_popnmbr ?? 0;
+
           const formattedPopNum = ipc_popnmbr.toLocaleString('en-US', {
             useGrouping: true,
           });
-          const title = `${countryName}`;
-          const body = `Date of analysis: ${date_of_analysis} <br> Validity period: ${analysis_period}`;
-          const conclusion = `${formattedPopNum} people in IPC/CH Phase 3 and above <br> (${ipc_percent}% of people in the analyzed areas)`;
-
-          const popupContent = this.generatePopupContent(title, body, conclusion);
+          const popupDetails = `Date of analysis: ${date_of_analysis} <br> Validity period: ${analysis_period}`;
+          const popupSummary = `
+          <span class="text-base font-semibold" style="color: #E08004;">${formattedPopNum}</span> 
```

Comment:
Avoid using the style attribute, use Tailwind instead. Just add the color to `tailwind.config.js` and add the class `text-yourAddedColorName`. (same below)

And I wonder why you are using a template string here instead of a React Element - is there a reason for this? I think removing the backticks and dollar signs and replacing `class` with `className` would do the same in a more React-like way.

Reply:
At first, I thought about that, but since the file was `.ts`, I couldn't append React components. I completely forgot to fix that, but now I have changed the file to `.tsx` thus I can safely append it.